### PR TITLE
feat: collapsable legs on travel search

### LINKED ISSUE #2114: [Bug] Når et reiseforslag blir langt blir det visuelt kuttet uten at det er mulighet for å scrolle horisontalt
### Beskrivelse
Når man får opp reiseforslag så vil visningen i flere tilfeller vise legs og overganger som går utover tilgjengelig plass. Dette feltet skal være scrollbart, men i praksis så er det vanskelig å scrolle uten å trykke på reisen slik at detaljene lastes. På Andorid kan man få det til om man er veldig rask med å scrolle etter å ha "trykket", mens på iPhone så er dette nærmest umulig.

Denne saken er omtalt på [Slack](https://mittatb.slack.com/archives/C0116FMPX4Y/p1643288458048400).
Gjelder både reisesøk v1 og v2.

### Løsning
Skjermbilde 2022-02-15 kl  15 59 45

- [ ] Ingen horisontal scroll. 
- [ ] Hele feltet er fortsatt klikkbart for å gå til detaljer
- [ ] Ved mange legs trunkerer vi og viser f. eks “+3” hvis det er 3 legs som er kuttet i visningen.
- [ ] Farge på "+3" er transport-other

### Implementasjon
Det er ikke rett fram å vite hvor mange legs som skal kollapses til en `+n`. forsøk på å lage en enkel cut-off på 3 eller 4 legs viste seg å ikke funke noe bra på veldig små skjermer eller for brukere med større skrifttype. Skrev derfor om til en dynamisk startegi, der vi prøver å rendre ut alle legs, for så å se om vi får plass. hvis vi ikke får plass kollapses en og en til vi er innefor bredden av komponenten. Det førte til litt hopping og lugging. Vi skjuler derfor komponenten til den er ferdig kollapset, og fader det inn. 
Fant samtidig at for brukere med stor skrift/liten skjerm, så brekker detaljer-knappen over 2 linjer. restrukturerte derfor result footer til å forkorte holdeplassnavn med "..." hvis det ikke er plass.
Litt sparring med @hildeor førte til at vi kutter ut mål-ikonet, siden det ikke egentlig gir noen verdi, og bare tar opp plass.

https://user-images.githubusercontent.com/1464594/154708665-a8a782e3-d220-420b-8d3b-9bb675d210bf.mp4

### Testbeskrivelse
jeg tror det viktigste her er å teste at "+n"-boblen teller riktig i alle tilfeller, evt være bevisst på om antallet gir mening for brukeren (burde venting telle?). 
Test på stor skrifttype, store og små skjermer.

### PR DESCRIPTION
**Pull request by monopoint**
fixes #2114

### CODE REVIEW

File: src/components/transportation-icon/index.tsx [hunk 1]
```diff
@@ -89,6 +89,28 @@ export function getTransportModeSvg(mode?: AnyMode) {
...
+      <View style={[styles.transportationIcon, {backgroundColor}]}>
+        <ThemeText
+          color={'transport_other'}
+          style={styles.lineNumberText}
+          type="body__primary--bold"
+        >
+          +{legs.length}
+        </ThemeText>
+      </View>
+    </>
```
Comment (rosvik) [inline src/components/transportation-icon/index.tsx:110]: Man trenger vel ikke å wrappe denne i "<></>"?

```suggestion
    <View style={[styles.transportationIcon, {backgroundColor}]}>
      <ThemeText
        color={'transport_other'}
        style={styles.lineNumberText}
        type="body__primary--bold"
      >
        +{legs.length}
      </ThemeText>
    </View>
```

File: src/screens/Assistant_v2/ResultItem.tsx [hunk 2]
```diff
@@ -284,7 +322,7 @@ const FootLeg = ({leg, nextLeg}: {leg: Leg; nextLeg?: Leg}) => {
       ) : (
         <ThemeIcon svg={WalkingPerson} />
       )}
-      <ThemeIcon svg={ChevronRight} size={'small'} />
+      {nextLeg && <ThemeIcon svg={ChevronRight} size={'small'} />}
```
Comment (rosvik) [inline src/screens/Assistant_v2/ResultItem.tsx:325]:
```suggestion
      {nextLeg && <ThemeIcon svg={ChevronRight} size="small" />}
```

File: src/screens/Assistant_v2/ResultItem.tsx [hunk 3]
```diff
@@ -202,11 +242,9 @@ const useThemeStyles = StyleSheet.createThemeHook((theme) => ({
     borderRadius: theme.border.radius.regular,
     marginTop: theme.spacings.medium,
   },
-  scrollContainer: {
-    padding: theme.spacings.medium,
-  },
+
```
Comment (rosvik) [inline src/screens/Assistant_v2/ResultItem.tsx:245]:
```suggestion
```

**Review by rosvik — APPROVED** [review]
Ser bra ut! Noen nitpicks bare.

File: src/components/transportation-icon/index.tsx (hunk 1 above)
Comment (monopoint) [reply src/components/transportation-icon/index.tsx:110]: :+1:

File: src/screens/Assistant_v2/ResultItem.tsx (hunk 2 above)
Comment (monopoint) [reply src/screens/Assistant_v2/ResultItem.tsx:325]: :+1:

**Pull request merged**